src/main/flight/pid.c

```diff
@@ -445,19 +446,63 @@ STATIC_UNIT_TESTED FAST_CODE_NOINLINE float pidLevel(int axis, const pidProfile_
+        const float rcCommandDelta = fabsf(getRcCommandDelta(axis));
+        float angleFeedforwardInput = pidRuntime.angleFeedforward[axis];
+        uint8_t duplicateCount = getFeedforwardDuplicateCount(axis);
+        if (rcCommandDelta) {
+            const float rxRateHz = 1e6f / getCurrentRxRefreshRate();
+            angleFeedforwardInput = (angle - pidRuntime.previousAngle[axis]) * rxRateHz;
+            //jitter attenuation
+            float jitterAttenuator = 1.0f;
+            if (feedforwardJitterFactor) {
+                if (rcCommandDelta < feedforwardJitterFactor) {
+                    jitterAttenuator = MAX(1.0f - (rcCommandDelta / feedforwardJitterFactor), 0.0f);
+                    jitterAttenuator = 1.0f - jitterAttenuator * jitterAttenuator;
+                }
+            }
+            angleFeedforwardInput *= jitterAttenuator;
+            //duplicate handling
+            if (duplicateCount) {
+                    angleFeedforwardInput /= duplicateCount + 1;
+                }
+        } else {
+            if (duplicateCount > 1) {
+                angleFeedforwardInput = 0;
+            }
+        }
+        //angle feedforward filter at RCrate
+        pidRuntime.angleFeedforward[axis] = pt2FilterApply(&pidRuntime.angleFeedforwardPt2[axis], angleFeedforwardInput);
+        pidRuntime.previousAngle[axis] = angle;
+    }
+    //angle feedforward filter at the PID loop rate
+    angleFeedforward = pt3FilterApply(&pidRuntime.angleFeedforwardPt3[axis], pidRuntime.angleFeedforward[axis]);
+    angleFeedforward *= pidRuntime.levelFeedforwardGain;
+#endif
     if (FLIGHT_MODE(ANGLE_MODE) || FLIGHT_MODE(GPS_RESCUE_MODE)) {
-        // ANGLE mode - control is angle based
-        const float setpointCorrection = errorAngle * pidRuntime.levelGain;
-        currentPidSetpoint = pt3FilterApply(&pidRuntime.attitudeFilter[axis], setpointCorrection);
+        // ANGLE mode - control is angle based with P term based on angle error and feedforward based on angle setpoint velocity
+        angleErrorCorrection *= levelErrorGainBoost;
+        rawSetpoint = angleErrorCorrection + angleFeedforward;
     } else {
         // HORIZON mode - mix of ANGLE and ACRO modes
         // mix in errorAngle to currentPidSetpoint to add a little auto-level feel
-        const float setpointCorrection = errorAngle * pidRuntime.horizonGain * horizonLevelStrength;
-        currentPidSetpoint += pt3FilterApply(&pidRuntime.attitudeFilter[axis], setpointCorrection);
+        rawSetpoint += (errorAngle + angleFeedforward) * horizonLevelStrength * pidRuntime.horizonGain;
     }
-    return currentPidSetpoint;
+    //logging
+    if (axis == FD_ROLL) {
+        DEBUG_SET(DEBUG_ANGLEMODE, 0, lrintf(angle * 10.0f));
+        DEBUG_SET(DEBUG_ANGLEMODE, 1, lrintf(angleErrorCorrection * 10.0f));
+        DEBUG_SET(DEBUG_ANGLEMODE, 2, lrintf(angleFeedforward * 10.0f));
+        DEBUG_SET(DEBUG_ANGLEMODE, 3, lrintf(levelErrorGainBoost * 100.0f));
```

Comment:
If, as per my other comment, we remove errorGainBoost altogether, we can use debug 3 for `currentAngle`.
It might then be better if we then re-arranged the order of the debugs so that they were in the following order:
- target angle
- current angle
- angle error correction
- angle feedforward correction